src/messaging/ExchangeMessageDispatch.cpp

```diff
@@ -80,7 +80,19 @@ CHIP_ERROR ExchangeMessageDispatch::SendMessage(SessionHandle session, uint16_t
         std::unique_ptr<ReliableMessageMgr::RetransTableEntry, decltype(deleter)> entryOwner(entry, deleter);
 
         ReturnErrorOnFailure(PrepareMessage(session, payloadHeader, std::move(message), entryOwner->retainedBuf));
-        ReturnErrorOnFailure(SendPreparedMessage(session, entryOwner->retainedBuf));
+        CHIP_ERROR err = SendPreparedMessage(session, entryOwner->retainedBuf);
+        if (err == System::MapErrorPOSIX(errno))
+        {
+            // sendmsg on BSD-based systems never blocks, no matter how the
+            // socket is configured, and will return ENOBUFS in situation in
+            // which Linux, for example, blocks.
+            //
+            // This is typically a transient situation, so we pretend like this
+            // packet drop happened somewhere on the network instead of inside
+            // sendmsg and will just resend it in the normal MRP way later.
```

Comment:
How frequently does this happen? Could we log as a warning at  least otherwise this is really hard to debug (silent error sending  something).

Reply:
It happens once every 20-40 test runs in my measurement attempts from earlier this summer.

I can absolutely log this.